byterun/caml/shared_heap.h

```diff
@@ -22,8 +22,10 @@ intnat caml_sweep(struct caml_heap_state*, intnat);
 /* must be called during STW */
 void caml_cycle_heap_stw(void);
 
-/* must be called on each domain 
+/* must be called on each domain
    (after caml_cycle_heap_stw) */
 void caml_cycle_heap(struct caml_heap_state*);
 
+int is_garbage (value);
```

Comment:
I don't really like this function, because it's not in general valid to check whether a value is garbage. The only correct use of this is `Assert(!is_garbage(...))`. How about adding `Check_hd(hd)` to `misc.h` instead, which checks that `hd != 0` and `hd` is not garbage?